services/cwmp-tr069.go

```diff
@@ -0,0 +1,165 @@
+/*
+* Honeytrap
+* Copyright (C) 2016-2017 DutchSec (https://dutchsec.com/)
+*
+* This program is free software; you can redistribute it and/or modify it under
+* the terms of the GNU Affero General Public License version 3 as published by the
+* Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+* details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* version 3 along with this program in the file "LICENSE".  If not, see
+* <http://www.gnu.org/licenses/agpl-3.0.txt>.
+*
+* See https://honeytrap.io/ for more details. All requests should be sent to
+* licensing@honeytrap.io
+*
+* The interactive user interfaces in modified source and object code versions
+* of this program must display Appropriate Legal Notices, as required under
+* Section 5 of the GNU Affero General Public License version 3.
+*
+* In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+* these Appropriate Legal Notices must retain the display of the "Powered by
+* Honeytrap" logo and retain the original copyright notice. If the display of the
+* logo is not reasonably feasible for technical reasons, the Appropriate Legal Notices
+* must display the words "Powered by Honeytrap" and retain the original copyright notice.
+ */
+package services
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+
+	"encoding/xml"
+	"github.com/honeytrap/honeytrap/event"
+	"github.com/honeytrap/honeytrap/pushers"
+)
+
+var (
+	_ = Register("cwmp", CWMP)
+)
+
+func CWMP(options ...ServicerFunc) Servicer {
+	s := &cwmpService{}
+
+	for _, o := range options {
+		o(s)
+	}
+
+	return s
+}
+
+type cwmpService struct {
+	c pushers.Channel
+}
+
+func (s *cwmpService) CanHandle(payload []byte) bool {
```

Comment:
Can we add here some more unique features to distinguish normal http, ethereum and this protocol. Check for the paths eg.

Reply:
Not much can be done for checking the paths, I'm afraid - GET requests might be indistinguishable from plain HTTP stuff. However, I improved the POST checking so that it checks for `<` (hinting at XML), `SOAP`/`soap` and `xml`.

Reply:
Not that familiair with the specifics, but there isn't a specific targetted path?

Reply:
I'm not familiar with the protocol either, but it seems that it doesn't have a specific path.

Quick example: the linked Wireshark capture uses `/UD/act?1`, [this article](http://djuro82.blogspot.it/2011/05/tr-069-cpe-provisioning.html) uses `/cwmpWeb/CPEMgt`, [this guide](http://support.huawei.com/enterprise/docinforeader!loadDocument1.action?contentId=DOC1000060529&partNo=100112) uses `/acs`.